include/logging/log_core.h

```diff
@@ -210,6 +211,26 @@ extern "C" {
 	)								    \
 	))
 
+#ifdef CONFIG_LOG_MINIMAL
+void log_minimal_hexdump_print(const char *data, size_t size);
+
+#define Z_LOG(_level, fmt, ...) do { \
```

Comment:
I would prefer that there is no such fork in definition. So far all options could be implemented is `if(cond)` in single macro and i'm afraid that it is an exception which will allow other options to implement that way and we might end up having multiple Z_LOG implementations.

I will look into that as it might not be as simple as such fork. Main issue is probably with instance logging.

Reply:
>  we might end up having multiple Z_LOG implementations.

What difference does that make?
Z_LOG is a private API...

Reply:
it's rather about maintenance. If new option appears that applies to minimal case as well then it will need to be added to 2 macros instead of one. So far logger supports multiple options and they could be handled in that if statement and minimal creates exception. As it is now, it could be done as well as:
```
#define __LOG()
   if (Z_LOG_CONST_LEVEL_CHECK(_level)) {
      if (IS_ENABLED(LOG_MINIMAL)) {
          ....
     } //rest of macro as it is now
     ....
```
That won't work out of the box, because some minor changes here and there are needed but would look cleaner.